### ISSUE TITLE
GooglePayButton is not clickable with Flutter 3.13

### ISSUE THREAD

**Issue by letsar** [flutter-stripe/flutter_stripe#1393]
After upgrading to Flutter 3.13 we can no longer click on `GooglePayButton`.

**To Reproduce**
1. Install Flutter 3.13
2. Run the code on this repo: https://github.com/flutter-stripe/flutter_stripe/tree/main/example
3. Expand the Wallets section and then click on Google Pay item
4. Click on the Google Pay button
5. Nothing happens

**Expected behavior**
We should be able to click on the Google Pay button.

**Smartphone / tablet**
 - Device: OnePlus 11
 - OS: Android 13
 - Package version: 9.3.0
 - Flutter version 3.13.2

**Additional context**
I think this is more a Flutter issue, since it worked in a previous version (tested in 3.7), and it's maybe linked to `PlatformViewLink`.
But in case there is a workaround, or you believe it's from this package, I created this issue.

**Comment by remonh87** [comment]
Can you try it with another device than the one plus? I did try it on an Google pixel4a and the callback is executed for me .

**Comment by letsar** [comment]
Hi @remonh87, we tried with at least two different phones. I asked for the other model. On which version is your Google Pixel 4a? Maybe it's also related to a specific Android version.

**Comment by remonh87** [comment]
Mine is a pixel4a on Android 13

**Comment by letsar** [comment]
The other phone where it does not work is a Xiaomi redmi note 11 with Android 12, so it's not because of the Android version since it works with yours which is on Android 13 like mine.

**Comment by leewrigg** [comment]
We're also experiencing this issue on Flutter 3.13

**Comment by remonh87** [comment]
@letsar  I see that you have contact with the flutter team.  I can help creating an example repo if needed but you have to test.

In the mean time can you run the example app (google pay Stripe in wallets) on branch `test-android-view` .  Maybe it is in the expensive androidview that we are using

**Comment by letsar** [comment]
Hi @remonh87, it would be really helpful if you could create an minimal reproducible repo which I could test, thank you very much.

I tested the branch `test-android-view` and I have the same issue, the button is not clickable.

**Comment by remonh87** [comment]
created: https://github.com/remonh87/platform_view_test . 

Can you also try it out by commenting out  `layout.initialize()` line 27 on `NativeView.kt` and press in the middle of the screen. My assumption is that it will work then and you will see pressed in the console

**Comment by letsar** [comment]
Thanks @remonh87!

So I tested the repo you created and it does nothing when I click on `NativeButton`. But as you predicted, I can see `tapped` in the console when I comment `layout.initialize()`. What is your assumption? Can we safely say that it's a Flutter issue?

**Comment by remonh87** [comment]
yes I think so I will comment on the Flutter issue that you created!

**Comment by letsar** [comment]
Thank you so much @remonh87, have a nice day.

**Comment by remonh87** [comment]
I had a hunch today and pushed a potential fix to `test-android-view` can you test it ?

**Comment by letsar** [comment]
Sadly it still doesn't work on this branch :-(

**Comment by jooikwanw** [comment]
I am having the same issue too. On the emulator and Flutter 3.13.1

I also tried `test-android-view` and the button does not do anything when pressed.

**Comment by tinrunje** [comment]
I can confirm this issue exists on both Xiaomi devices and One Plus devices. Even if you just attempt to call the bottom sheet to insert a card it will not appear on those devices.

**Comment by manandhar-aarya** [comment]
I am facing this issue with Flutter 3.16.3 too. 

Flutter doctor logs:

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.16.3, on macOS 14.1.2 23B92 darwin-arm64, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] VS Code (version 1.80.2)
[✓] Connected device (3 available)
[✓] Network resources


dependencies:
  stripe_js: 3.3.0
  flutter_stripe: 9.6.0
  flutter_stripe_web: 4.6.0

**Comment by ziagit** [comment]
I'm having same issue. 
It was working in flutter 3.10 after upgrade to 3.16 it does not working. 
I tested on Samsung A12 Android 12.

Please fix this issue.

**Comment by ziagit** [comment]
> I am facing this issue with Flutter 3.16.3 too with device Samsung Galaxy A13, Android Version 13
> 
> Flutter doctor logs:
> 
> Doctor summary (to see all details, run flutter doctor -v): [✓] Flutter (Channel stable, 3.16.3, on macOS 14.1.2 23B92 darwin-arm64, locale en-GB) [✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0) [✓] Xcode - develop for iOS and macOS (Xcode 15.1) [✓] Chrome - develop for the web [✓] Android Studio (version 2022.2) [✓] VS Code (version 1.80.2) [✓] Connected device (3 available) [✓] Network resources
> 
> dependencies: stripe_js: 3.3.0 flutter_stripe: 9.6.0 flutter_stripe_web: 4.6.0
> 
> I downgraded flutter to version 3.10.4 for test and then it works. I cannot downgrade to 3.10.4 at the moment because I want to use stripe-express and the latest version of flutter_stripe.

@manandhar-aarya could you solve this issue?

**Comment by manandhar-aarya** [comment]
> > I am facing this issue with Flutter 3.16.3 too with device Samsung Galaxy A13, Android Version 13
> > Flutter doctor logs:
> > Doctor summary (to see all details, run flutter doctor -v): [✓] Flutter (Channel stable, 3.16.3, on macOS 14.1.2 23B92 darwin-arm64, locale en-GB) [✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0) [✓] Xcode - develop for iOS and macOS (Xcode 15.1) [✓] Chrome - develop for the web [✓] Android Studio (version 2022.2) [✓] VS Code (version 1.80.2) [✓] Connected device (3 available) [✓] Network resources
> > dependencies: stripe_js: 3.3.0 flutter_stripe: 9.6.0 flutter_stripe_web: 4.6.0
> > I downgraded flutter to version 3.10.4 for test and then it works. I cannot downgrade to 3.10.4 at the moment because I want to use stripe-express and the latest version of flutter_stripe.
> 
> @manandhar-aarya could you solve this issue?

no, for now i am using flutter v3.10 for android and v3.13 for web and ios.

**Comment by ziagit** [comment]
> > > I am facing this issue with Flutter 3.16.3 too with device Samsung Galaxy A13, Android Version 13
> > > Flutter doctor logs:
> > > Doctor summary (to see all details, run flutter doctor -v): [✓] Flutter (Channel stable, 3.16.3, on macOS 14.1.2 23B92 darwin-arm64, locale en-GB) [✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0) [✓] Xcode - develop for iOS and macOS (Xcode 15.1) [✓] Chrome - develop for the web [✓] Android Studio (version 2022.2) [✓] VS Code (version 1.80.2) [✓] Connected device (3 available) [✓] Network resources
> > > dependencies: stripe_js: 3.3.0 flutter_stripe: 9.6.0 flutter_stripe_web: 4.6.0
> > > I downgraded flutter to version 3.10.4 for test and then it works. I cannot downgrade to 3.10.4 at the moment because I want to use stripe-express and the latest version of flutter_stripe.
> > 
> > 
> > @manandhar-aarya could you solve this issue?
> 
> no, for now i am using flutter v3.10 for android and v3.13 for web and ios.

Ok,
What version of flutter_stripe are you using with flutter 3.10?
Have you tested with Android above 12?

**Comment by manandhar-aarya** [comment]
these are the versions i am using: 

  stripe_js: 3.3.0
  flutter_stripe: 9.5.0+1
  flutter_stripe_web: 4.5.0

and yes i tried on android 13 and it seems to work.

**Comment by remonh87** [comment]
The problem is that there is regression from Flutter engine and there is little we can do ourselves about it. The paybutton is made by Stripe and the issue is in the flutter engine.  As temp work around I would recommend: https://github.com/flutter-stripe/flutter_stripe/blob/main/example/lib/screens/wallets/google_pay_screen.dart or use the paymentsheet . I will ask some help from the stripe team but I do not expect this issue to be solved quickly.

**Comment by ziagit** [comment]
> these are the versions i am using:
> 
> stripe_js: 3.3.0 flutter_stripe: 9.5.0+1 flutter_stripe_web: 4.5.0
> 
> and yes i tried on android 13 and it seems to work.

After downgrade to 3.10 I get this error

```
LocalizedErrorMessage(code: FailureCode.Failed, localizedMessage: The Android manifest needs to be updated to use Google Pay API
```

but my AndroidManifest.xml is ok. It was working on 3.7

Part of `AndroidManifest.xml`
```
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
   <application
        <activity
            android:name=".MainActivity"
            android:theme="@style/LaunchTheme"

            <!--Google pay-->
            <meta-data
                android:name="com.google.android.gms.wallet.api.enabled"
                android:value="true"/>
```

**Comment by ziagit** [comment]
> The problem is that there is regression from Flutter engine and there is little we can do ourselves about it. The paybutton is made by Stripe and the issue is in the flutter engine. As temp work around I would recommend: https://github.com/flutter-stripe/flutter_stripe/blob/main/example/lib/screens/wallets/google_pay_screen.dart or use the paymentsheet . I will ask some help from the stripe team but I do not expect this issue to be solved quickly.

Didn't you solve this issue in v10.0.0 without using pay plugin?

**Comment by urielvexi** [comment]
> > The problem is that there is regression from Flutter engine and there is little we can do ourselves about it. The paybutton is made by Stripe and the issue is in the flutter engine. As temp work around I would recommend: https://github.com/flutter-stripe/flutter_stripe/blob/main/example/lib/screens/wallets/google_pay_screen.dart or use the paymentsheet . I will ask some help from the stripe team but I do not expect this issue to be solved quickly.
> 
> Didn't you solve this issue in v10.0.0. Cannot use pay plugin.

other work around could be to use:
```
Stack(
    children: [
        Stripe.PlatformPayButton(
            onPressed: () {},
        ),
        InkWell(
            onTap: controller.payWithGooglePay,
            child: Container(),
        ),
    ],
);
```

instead of:
```
Stripe.PlatformPayButton(
    onPressed: controller.payWithGooglePay,
),
```

**Comment by ziagit** [comment]
> these are the versions i am using:
> 
> stripe_js: 3.3.0 flutter_stripe: 9.5.0+1 flutter_stripe_web: 4.5.0
> 
> and yes i tried on android 13 and it seems to work.

Do you use as I do? because my code works only on android 12 and below

```
 @override
  Widget build(BuildContext context) {
    if (defaultTargetPlatform == TargetPlatform.android)
      return Padding(
        padding: const EdgeInsets.fromLTRB(14, 8, 14, 8),
        child: SizedBox(
          height: 65,
          child: PlatformPayButton(
            onPressed: _handlePayPress,
          ),
        ),
      );
    else
      return SizedBox();
  }
  
   Future<void> _handlePayPress() async {
   print("Google pay button pressed!");
}
```

**Comment by ziagit** [comment]
> > > The problem is that there is regression from Flutter engine and there is little we can do ourselves about it. The paybutton is made by Stripe and the issue is in the flutter engine. As temp work around I would recommend: https://github.com/flutter-stripe/flutter_stripe/blob/main/example/lib/screens/wallets/google_pay_screen.dart or use the paymentsheet . I will ask some help from the stripe team but I do not expect this issue to be solved quickly.
> > 
> > 
> > Didn't you solve this issue in v10.0.0. Cannot use pay plugin.
> 
> other work around could be to use:
> 
> ```
> Stack(
>     children: [
>         Stripe.PlatformPayButton(
>             onPressed: () {},
>         ),
>         InkWell(
>             onTap: controller.payWithGooglePay,
>             child: Container(),
>         ),
>     ],
> );
> ```
> 
> instead of:
> 
> ```
> Stripe.PlatformPayButton(
>     onPressed: controller.payWithGooglePay,
> ),
> ```

Have you tested?

**Comment by ziagit** [comment]
For me GPay button worked only on Android 11 and 12 but in Android 7 and 14 I tested the button is not only clickable it does not taking it's design to.

I found a workaround solution by creating a custom GPay button and call _handleOnPress()

```
 @override
  Widget build(BuildContext context) {
    if (defaultTargetPlatform == TargetPlatform.android &&
        Stripe.instance.isPlatformPaySupportedListenable.value)
      return InkWell(
        child: Container(
          height: 54,
          margin: EdgeInsets.all(4),
          decoration: BoxDecoration(
              color: Colors.black, borderRadius: BorderRadius.circular(4)),
          child: Row(
            crossAxisAlignment: CrossAxisAlignment.center,
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              SizedBox(width: 24, child: assetImage(context, "google.png")),
              SizedBox(width: 4),
              Text("Pay", style: TextStyle(color: Colors.white, fontSize: 21))
            ],
          ),
        ),
        onTap: () => _handlePayPress(),
      );
    else
      return SizedBox();
  }

  Future<void> _handlePayPress() async {
    String price = (widget.order.mover.price).toStringAsFixed(2);
    String sanitizedValue = price.replaceAll(RegExp(r'[^\d.]'), '');
    double doubleValue = double.parse(sanitizedValue);
    int constPrice = (doubleValue * 100).toInt();
    try {
      final paymentMethod =
          await Stripe.instance.createPlatformPayPaymentMethod(
        params: PlatformPayPaymentMethodParams.googlePay(
          googlePayParams: GooglePayParams(
            merchantCountryCode: "CA",
            currencyCode: "CAD",
            merchantName: "Tingsapp",
            isEmailRequired: true,
            testEnv: false,
          ),
          googlePayPaymentMethodParams: GooglePayPaymentMethodParams(
            amount: constPrice,
            billingAddressConfig:
                GooglePayBillingAddressConfig(isRequired: true),
            shippingAddressConfig:
                GooglePayShippingAddressConfig(isRequired: true),
          ),
        ),
      );
      handlePaymentMethod(context, paymentMethod);
    } catch (e) {
      if (e is StripeException) {
        gpayResponse = "${e.error}";
      } else {
        gpayResponse = "${e}";
      }
      setState(() {});
    }
  }
```

**Comment by PeterMX** [comment]
> > > The problem is that there is regression from Flutter engine and there is little we can do ourselves about it. The paybutton is made by Stripe and the issue is in the flutter engine. As temp work around I would recommend: https://github.com/flutter-stripe/flutter_stripe/blob/main/example/lib/screens/wallets/google_pay_screen.dart or use the paymentsheet . I will ask some help from the stripe team but I do not expect this issue to be solved quickly.
> > 
> > 
> > Didn't you solve this issue in v10.0.0. Cannot use pay plugin.
> 
> other work around could be to use:
> 
> ```
> Stack(
>     children: [
>         Stripe.PlatformPayButton(
>             onPressed: () {},
>         ),
>         InkWell(
>             onTap: controller.payWithGooglePay,
>             child: Container(),
>         ),
>     ],
> );
> ```
> 
> instead of:
> 
> ```
> Stripe.PlatformPayButton(
>     onPressed: controller.payWithGooglePay,
> ),
> ```

@ziagit I did test it, looks like it works:
<p float="left">
<img src="https://apigow.urieldev.com/assets/gh_issue_35.jpeg" width=300>
<img src="https://apigow.urieldev.com/assets/gh_issue_36.jpeg" width=300>
</p>

about the button design, I did a PR #1566 to fix it, I don't know if it is the design you are talking about

**Comment by ziagit** [comment]
> InkWell(
>             onTap: controller.payWithGooglePay,
>             child: Container(),
>         ),

How declared `controller` ?

I created a custom button and calling _onPayPress() function like below

```
return InkWell(
        child: Container(
          height: 54,
          margin: EdgeInsets.all(4),
          decoration: BoxDecoration(
              color: Colors.black, borderRadius: BorderRadius.circular(4)),
          child: Row(
            crossAxisAlignment: CrossAxisAlignment.center,
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              SizedBox(width: 24, child: assetImage(context, "google.png")),
              SizedBox(width: 4),
              Text("Pay", style: TextStyle(color: Colors.white, fontSize: 21))
            ],
          ),
        ),
        onTap: () => _handlePayPress(),
      );
```

It works in Test mode but in live mode I get Authentication Error when I click the button. Don't you?

Could you share how you defined the `controller`?

**Comment by leetcoder314** [comment]
> The problem is that there is regression from Flutter engine and there is little we can do ourselves about it. The paybutton is made by Stripe and the issue is in the flutter engine. As temp work around I would recommend: https://github.com/flutter-stripe/flutter_stripe/blob/main/example/lib/screens/wallets/google_pay_screen.dart or use the paymentsheet . I will ask some help from the stripe team but I do not expect this issue to be solved quickly.

What is the difference between google_pay_screen.dart vs google_pay_stripe_screen.dart?

**Comment by rahulrmishra** [comment]
Any updates on this or by when will this be resolved? Is anyone looking into the issue?
